main.go

```diff
@@ -88,6 +89,11 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
+	restConfig := ctrl.GetConfigOrDie()
+
+	// Get LeaderElection configs
+	leConfig := leaderelection.GetLeaderElectionConfig(restConfig, enableLeaderElection)
```

Comment:
This func could also be eliminated.. something to consider using in the future something this style

```
	builder := NewController(c.componentName, c.startFunc).
		WithKubeConfigFile(c.basicFlags.KubeConfigFile, nil).
		WithComponentNamespace(c.basicFlags.Namespace).
		WithLeaderElection(config.LeaderElection, c.basicFlags.Namespace, c.componentName+"-lock").
		WithVersion(c.version).
		WithHealthChecks(c.healthChecks...).
		WithEventRecorderOptions(events.RecommendedClusterSingletonCorrelatorOptions()).
		WithRestartOnChange(exitOnChangeReactorCh, startingFileContent, observedFiles...).
		WithComponentOwnerReference(c.ComponentOwnerReference)
```
https://github.com/openshift/library-go/blob/master/pkg/controller/controllercmd/cmd.go#L316C1-L324C57